db/migrate/20250303175124_add_default_period_to_users.rb

```diff
@@ -0,0 +1,5 @@
+class AddDefaultPeriodToUsers < ActiveRecord::Migration[7.2]
+  def change
+    add_column :users, :default_period, :string
```

Comment:
Since we're now setting this across several views, I'd be in favor of setting a default of `last_30_days` and then adding a controller concern to set it for each request (ensuring it _always_ has a valid value):

```rb
module Periodable
  extend ActiveSupport::Concern 

  included do
    before_action :set_period 
  end

  private
    def set_period
      @period = Period.from_key(params[:period] || Current.user&.default_period)
    rescue Period::InvalidKeyError
      @period = Period.last_30_days
    end
end
```